### PR TITLE
271 update use cases

### PR DESCRIPTION
**Pull request by Nid21cs**
Added use cases for features that Profbook has

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -455,26 +637,153 @@ otherwise)
...
+    * 2b1. ProfBook shows an error message.
 
       Use case resumes at step 1.
 
-*{More to be added}*
+**Use case: Create a group**
+
+**MSS**
+
+1. User requests to create a group witb path and name
```
Comment (mingyuanc) [inline docs/DeveloperGuide.md:663]: should it be with?

**Review by mingyuanc — DISMISSED** [review]
Fix the spelling error then you are all good to merge! LGTM

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -434,12 +434,194 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+    * 2b1. ProfBook shows list of children under current directory.
+
+      Use case ends.
+
+**Use case: View tasks under a specified path**
+
+**MSS**
+
+1. User requests to view tasks under specified path, either student or group.
```
Comment (zannloo) [inline docs/DeveloperGuide.md:512]: User is able to use the cat command with and without specifying a path. Perhaps could add more details like the use case for ls above.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -434,12 +434,194 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+2. ProfBook stores all data and closes application.
+
+   Use case ends.
+
+**Use case: Add a student**
+
+**MSS**
+
+1. User requests to add a student .
+2. AddressBook deletes the person.
```
Comment (zannloo) [inline docs/DeveloperGuide.md:548]: Not sure if this is still relevant.

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -434,12 +434,194 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. User requests to exit ProfBook
+2. ProfBook stores all data and closes application.
+
+   Use case ends.
+
+**Use case: Add a student**
+
+**MSS**
+
+1. User requests to add a student .
```
Comment (zannloo) [inline docs/DeveloperGuide.md:547]: There is an extra spacing between student and the fullstop.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -434,12 +434,194 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+**Extensions**
+
+* 2a. The given parameters are invalid.
+
+    * 2a1. ProfBook shows an error message.
+
+      Use case resumes at step 1.
+
+**Use case: Edit a students details**
```
Comment (zannloo) [inline docs/DeveloperGuide.md:560]: Not sure if it should be "Edit a student's details" instead.

Comment (zannloo) [inline docs/DeveloperGuide.md:560]: Not sure if it should be "Edit a student's details" instead.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -434,12 +434,194 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+      Use case resumes at step 1.
+
+* 2b. The given parameter is invalid.
+
+    * 2b1. ProfBook shows an error message.
+
+      Use case resumes at step 1.
+
+**Use case: Edit a groups details**
```
Comment (zannloo) [inline docs/DeveloperGuide.md:583]: Not sure if it should be "Edit a group's details" instead.

**Review by zannloo — DISMISSED** [review]
LGTM, very neat work with just a few minor adjustments that may need to be made. Good job!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Nid21cs) [reply docs/DeveloperGuide.md]: Added a extension for this thank you

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Nid21cs) [reply docs/DeveloperGuide.md]: done!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Nid21cs) [reply docs/DeveloperGuide.md]: updated

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (Nid21cs) [reply docs/DeveloperGuide.md]: updated

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (Nid21cs) [reply docs/DeveloperGuide.md]: done!

Comment (Nid21cs) [reply docs/DeveloperGuide.md]: Done!

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (Nid21cs) [reply docs/DeveloperGuide.md]: done!

**Review by mingyuanc — DISMISSED** [review]
LGTM

**Review by NereusWB922 — APPROVED** [review]
LGTM

**Pull request merged**